auraed/src/cells/cell_service/cells/cell_graph.rs

```diff
@@ -0,0 +1,24 @@
+use super::{CellName, CellSpec};
+
+#[derive(Clone, Debug)]
+pub struct GraphNode {
+    pub cell_info: Option<(CellName, CellSpec)>,
+    pub children: Vec<GraphNode>,
+}
+
+impl GraphNode {
+    pub fn with_cell_info(
+        self,
+        cell_name: &CellName,
+        cell_spec: &CellSpec,
```

Comment:
The function requires ownership of `cell_name` and `cell_spec`, so shouldn't hide that, IMO. I would clone at the call location if needed.